pilot/pkg/model/push_context.go

```diff
@@ -212,6 +212,7 @@ type PushContext struct {
 
 	// cache gateways addresses for each network
 	// this is mainly used for kubernetes multi-cluster scenario
+	networksMu      sync.RWMutex
```

Comment:
Why do we need the lock here, previously we do not have a chance to read/write concurrently

Reply:
Reading this in tests can cause a race there.

Reply:
good